yt_dlp/extractor/bilibili.py

```diff
@@ -381,20 +406,41 @@ def _real_extract(self, url):
             **info, **top_level_info
         }
 
-    def _extract_anthology_entries(self, bv_id, video_id, webpage):
-        title = self._html_search_regex(
-            (r'<h1[^>]+\btitle=(["\'])(?P<title>(?:(?!\1).)+)\1',
-             r'(?s)<h1[^>]*>(?P<title>.+?)</h1>',
-             r'<title>(?P<title>.+?)</title>'), webpage, 'title',
-            group='title')
-        json_data = self._download_json(
-            f'https://api.bilibili.com/x/player/pagelist?bvid={bv_id}&jsonp=jsonp',
-            video_id, note='Extracting videos in anthology')
-
-        if json_data['data']:
-            return self.playlist_from_matches(
-                json_data['data'], bv_id, title, ie=BiliBiliIE.ie_key(),
-                getter=lambda entry: 'https://www.bilibili.com/video/%s?p=%d' % (bv_id, entry['page']))
+    def _extract_anthology_entries(self, video_id, webpage):
+        json__INITIAL_STATE__ = self._parse_json(
+            self._search_regex(r'window.__INITIAL_STATE__\s*=\s*({.+?});\(function\(\)', webpage,
+                               '__INITIAL_STATE__', default=None) or '{}', video_id, fatal=False)
+
+        url_smuggled_data = '#__youtubedl_smuggle={"from_playlist":true}'
+
+        if 'sectionsInfo' in json__INITIAL_STATE__:
+            season_info = json__INITIAL_STATE__['sectionsInfo']
+
+            entries = [{
+                '_type': 'url_transparent',
+                'url': 'https://www.bilibili.com/video/' + episode['bvid'] + url_smuggled_data,
+                'ie_key': BiliBiliIE.ie_key(),
+                'timestamp': None,
+                'episode': episode.get('title')
+            } for episode in traverse_obj(season_info, ('sections', 0, 'episodes'))]
+
+            return self.playlist_result(entries, season_info['id'], season_info['title'])
+
+        elif 'epInfo' in json__INITIAL_STATE__:
+            bangumi_id = traverse_obj(json__INITIAL_STATE__, ('epInfo', 'aid'))
+            season_info = json__INITIAL_STATE__['mediaInfo']
+
+            entries = [{
+                '_type': 'url_transparent',
+                'url': episode['link'] + url_smuggled_data,
+                'ie_key': BiliBiliIE.ie_key(),
+                'timestamp': None,
+                'episode': episode.get('long_title')
+            } for episode in season_info['episodes']]
+
+            return self.playlist_result(entries, bangumi_id, season_info['season_title'])
+        else:
+            raise ExtractorError('Unknown __INITIAL_STATE__', expected=True, video_id=id)
```

Comment:
why expected?

Reply:
So this shouldn't have `expected=True`, right?

Reply:
> So this shouldn't have `expected=True`, right?

I think you are right, I dont realy know the meaning of it